r4e-img-upgrade-iso.sh

```diff
@@ -0,0 +1,59 @@
+# script for creating an update ISO
+# the image includes the updated agent RPM that communicates with server that monitors edge devices
+# at the end an upgraded ostree image will be available
+# input:
+#  - original-image-name that was chosen for the first step (Build ISO for edge devices)
+#  - parrent-ostree-commit can be found in the original image webserver under in compose.json file under the key 'ostree-commit'
+#  - image-version-number the new version of the image
+# script assumes that file `blueprint.toml` is present and updated in the working directory
+set -e
+Usage() {
+    echo "Usage:"
+    echo `basename $0` "<original-image-name> <parrent-ostree-commit> <image-version-number>"
+}
+
+# check usage
+if [ $# != 3 ] ; then
+    echo -e "bad usage\n"
+    Usage
+    exit 1
+fi
+
+# read input
+export IMAGE_NAME=$1
+export OSTREE_COMMIT=$2
+export IMAGE_VERSION=$3
+export NEW_IMAGE_NAME=${IMAGE_NAME}-${IMAGE_VERSION}
+
+# prepare local variables
+IMAGE_FOLDER=/var/www/html/${NEW_IMAGE_NAME}
+BLUEPRINT_FILE=blueprint.toml
+BLUPRINT_NAME=${IMAGE_NAME}
+
+
+# create a new blueprint
+composer-cli blueprints push $BLUEPRINT_FILE
+
+# validate blueprint
+echo "validating blueprint $BLUPRINT_NAME"
+composer-cli blueprints depsolve $BLUPRINT_NAME
+
+# build image using the composer based on the parent commit create a new commit- first find the parent commit hash in http://<host-ip>/<image-name>/compose.json under "ostree-commit"
+echo "creating a new commit from parrent $OSTREE_COMMIT using blueprint $BLUPRINT_NAME"
+composer-cli compose start-ostree $BLUPRINT_NAME rhel-edge-commit --parent $OSTREE_COMMIT > temp_upgrade.out
+BUILD_ID=`cat temp_upgrade.out | awk '{print $2}'`
+echo "waiting for build $BUILD_ID to be ready..."
+while [ "`composer-cli compose status | grep $BUILD_ID | grep -c RUNNING`" != "0" ] ; do sleep 5 ; done
+if [ "`composer-cli compose status | grep $BUILD_ID | grep -c FINISHED`" == "0" ] ; then
+    echo "image composition failed"
+    echo "check 'composer-cli compose status'"
+    exit 1
+fi
+
+
+# saving image to web folder
+echo "saving the new image to web folder "
```

Comment:
pls include the name of that folder in this message, so the user can refer to it from the kickstart file / beaker job if/when needed. Ideally, we could provide here a link to the http server that exposes the new commit, so the user can just copy it and paste it on the device ostree conf.